### PR TITLE
[BugFix] fix mv with count distinct rewrite bug

### LINKED ISSUE #16558: mv with count(distinct) on distributed column rewrite failed
### Steps to reproduce the behavior (Required)
1. create table and mv
create table emps (\n" +
                "    empid int not null,\n" +
                "    deptno int not null,\n" +
                "    name varchar(25) not null,\n" +
                "    salary double\n" +
                ")\n" +
                "distributed by hash(`empid`) buckets 10\n" +
                "properties (\n" +
                "\"replication_num\" = \"1\"\n" +
                ");

create materialized view mv_1 distributed by hash(`empid`) as select deptno, count(distinct empid) from emps group by deptno;

2. query
select deptno, count(distinct empid) from emps group by deptno;

3. check the plan whether use the mv_1

### Expected behavior (Required)
use mv_1
### Real behavior (Required)
not use mv_1
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
main

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #16558 

## Problem Summary(Required) ：

fix mv with count distinct rewrite bug. the bug is that mv rewrite conflicts with TF_REWRITE_GROUP_BY_COUNT_DISTINCT rule which generates agg-agg-scan pattern plan.
 mv rewriter does not support this pattern. fix it by adding new agg-agg-scan pattern rewrite rule.
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/rule/AggregateAggregateScanRule.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.rule.RuleType;
+import com.starrocks.sql.optimizer.rule.transformation.materialization.AggregatedMaterializedViewRewriter;
+import com.starrocks.sql.optimizer.rule.transformation.materialization.MaterializedViewRewriter;
+import com.starrocks.sql.optimizer.rule.transformation.materialization.MvUtils;
+
+/*
+ *
+ * Here is the rule for pattern Aggregate-Scan
```
[the remark this answers is not in the document]
Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/rule/AggregateAggregateScanRule.java]: done

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 17 (94.12%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/rule/AggregateAggregateScanRule.java|8|9|88.89%|[48]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleSet.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/AggregatedMaterializedViewRewriter.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java|5|5|100.00%|[]|

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by mofeiatwork** [comment]
run starrocks_admit_test

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**